python/tk_multi_publish2/dialog.py

```diff
@@ -1405,7 +1410,7 @@ def _on_item_context_change(self, context):
         # ...
         # for any items that have a new context, rerun plugin attachment so that
         # they are published using the destination context's plugins
-        # self._publish_manager.attach_plugins(items_with_new_context)
+        self._publish_manager._attach_plugins(items_with_new_context)
```

Comment:
This already happens as part of the set_context method on the item for the item and all its children, no need to run this again.

Reply:
I just realized there is an issue with the code in _set_context_r() so that on_context_changed() only runs for child items that don't already have their context set. That should be updated to run on_context_changed() always, which is probably why you thought this was necessary. I'll fix that tomorrow, or you can as part of this PR if you want.

Reply:
Yeah had it disabled at first, but then added at it back later. 
Instead of that check I will add, if the child context is different than the new one.

Reply:
Actually, I take that back, the logic in `_set_context_r()` is correct. If the child item has a local context override, it should not be updated.